client/my-sites/email/email-forwarding/email-forwarding-add-new-compact-list.jsx

```diff
@@ -0,0 +1,167 @@
+import { Button } from '@automattic/components';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import { validateAllFields } from 'calypso/lib/domains/email-forwarding';
+import EmailForwardingAddNewCompact from 'calypso/my-sites/email/email-forwarding/email-forwarding-add-new-compact';
+import { composeAnalytics, withAnalytics } from 'calypso/state/analytics/actions';
+import { addEmailForward } from 'calypso/state/email-forwarding/actions';
+import getEmailForwardingLimit from 'calypso/state/selectors/get-email-forwarding-limit';
+import {
+	addEmailForwardSuccess,
+	getEmailForwards,
+	isAddingEmailForward,
+} from 'calypso/state/selectors/get-email-forwards';
+import { isRequestingSiteDomains } from 'calypso/state/sites/domains/selectors';
+import { getSiteSlug } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+class EmailForwardingAddNewCompactList extends Component {
+	static propTypes = {
+		emailForwards: PropTypes.array,
+		emailForwardingLimit: PropTypes.number,
+		onConfirmEmailForwarding: PropTypes.func.isRequired,
+		onAddForwardSuccess: PropTypes.func,
+		selectedDomainName: PropTypes.string.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			emailForwards: [ { destination: '', mailbox: '', isValid: false } ],
+			isRedirecting: false,
+			newForwardAdded: false,
+		};
+	}
+
+	hasValidEmailForwards() {
+		return ! this.state.emailForwards?.some( ( forward ) => ! forward.isValid );
+	}
+
+	addNewEmailForwardWithAnalytics = ( domainName, mailbox, destination ) => {
+		withAnalytics(
+			composeAnalytics(
+				this.props.onConfirmEmailForwarding(),
+				this.props.addEmailForward( domainName, mailbox, destination )
+			)
+		);
+	};
+
+	addNewEmailForwardsClick = ( event ) => {
+		const { selectedSiteSlug, selectedDomainName } = this.props;
+
+		event.preventDefault();
+
+		if ( this.props.isAddingEmailForward ) {
```

Comment:
We could destructure that variable.

Reply:
I ended up renaming the prop and destructuring because the previous name clashed with an imported function.